### ISSUE TITLE
Reduce the size of PHPIDS core

### ISSUE THREAD

**Issue by lstrojny** [PHPIDS/PHPIDS#6]
While PHPIDS should still work as easy as possible out of the box it should as well integrate easily with third-party frameworks and existing software already providing infrastructure services like caching, reading configuration files and so on. Current PHPIDS tries to do all of that.
- Get rid of `IDS_Init`
- Get mostly rid of `IDS_Config_*`, only support a PHP config file and support a bridges for framework integration
- Get mostly rid of `IDS_Cache_*` and only support a simple file based caching layer. Provide bridges to other caching solutions
- Get rid of `IDS_Log_*` and support bridges to existing loggers
- Expose middleware for wider spread frameworks

**Comment by lstrojny** [comment]
See https://github.com/PHPIDS/PHPIDS/wiki/Version-2.0 for long term plans.

**Comment by kingcrunch** [comment]
Regarding `Init`: Should the config directly go into `Monitor`, or should it be replaced with a proper `Config`-class?

```
class Config {
  public $base_path = 'default';
}
```

Regarding `Caching`: A good start would be `PSR3` :smile:

**Comment by lstrojny** [comment]
That very much depends how huge the configuration will be. My rule of thumb: more than three related values go into a object.

**Comment by kingcrunch** [comment]
Is it correct, that logging is completely unused? I can't find any usage, except I use it directly myself of course. Just a suggestion, but couldn't it get simply removed?

**Comment by lstrojny** [comment]
Nice find, you are right. So we can get rid of it for now. I would think logging is a good idea, as long as it is decoupled from the core.

**Comment by kingcrunch** [comment]
I'm not sure... I don't know, what should be logged except the overall result (`Report`), but because the result is the reason why an application integrates PHPIDS it will likely take care of it itself including logging. Maybe to simplify logging in simpler setup a `__toString()` on `Report` is enough?

TL;DR: I have no idea what to _useful_ log where :smile:

**Comment by lstrojny** [comment]
There could be two default offerings for logging: a request log alike format and a more verbose logging format for serious audits. But that’s nothing to worry for now.

**Comment by jayzeng** [comment]
I have some ideas in mind (as I mentioned in email to @lstrojny), I am going to implement them in my fork. Will keep you guys posted when I have something to show.
